proposal/package-importer.d.ts.md

```diff
@@ -0,0 +1,382 @@
+known conflicts in the ecosystem.
+
+#### No built-in `pkg` resolver for browsers
+
+Dart Sass will not provide a built-in resolver for browsers to use the `pkg`
+scheme. To support a similar functionality, a user would need to ensure that
+files are served, and the loader would need to fetch the URL. In order to follow
+the same algorithm for [resolving a file: URL], we would need to make many
+fetches. If we instead require the browser version to have a fully resolved URL,
+we negate many of this spec's benefits. Users may write their own custom
+importers to fit their needs.
+
+[resolving a file: url]: ../spec/modules.md#resolving-a-file-url
+
+#### Available as an opt-in setting
+
+The `pkg` import loader will be exposed through an opt-in setting as it adds the
+potential for file system interaction to `compileString` and
+`compileStringAsync`. Specifically, we want people who invoke Sass compilation
+functions to have control over what files get accessed, and there's even a risk
+of leaking file contents in error messages.
+
+#### Node Resolution Decisions
+
+The current recommendation for resolving packages in Node is to add
+`node_modules` to the load paths. We could add `node_modules` to the load paths
+by default, but that lacks clarity to the implementation and the reader. In
+addition, a file may have access to multiple `node_modules` directories, and
+different files may have access to different `node_modules` directories in the
+same compilation.
+
+There are a variety of methods currently in use for specifying a location of the
+default Sass export for npm packages. For the most part, packages contain both
+JavaScript and styles, and use the `main` or `module` root keys to define the
+JavaScript entry point. Some packages use the `"sass"` key at the root of their
+`package.json`.
+
+Other packages have adopted [conditional exports], driven by build tools like
+[Vite], [Parcel] and [Sass Loader for Webpack] which all resolve Sass paths
+using the `"sass"` and the `"style"` custom conditions.
+
+[conditional exports]: https://nodejs.org/api/packages.html#conditional-exports
+[Vite]: https://github.com/vitejs/vite/pull/7817
+[Parcel]:
+    https://github.com/parcel-bundler/parcel/blob/2d2400ded4615375ee6bd53ef77b4857ad1591dd/packages/transformers/sass/src/SassTransformer.js#L163
+[Sass Loader for Webpack]:
+    https://github.com/webpack-contrib/sass-loader/blob/02df41203adfda96959e56abb43bd35a89ec11ba/src/utils.js#L514
+
+Because use of conditional exports is flexible and recommended for modern
+packages, this will be the primary method used for the Node package resolver. We
+will support both the `"sass"` and the `"style"` conditions, as Sass can also
+use the CSS exports exposed through `"style"`. While in practice, `"style"`
+tends to be used solely for `css` files, we will support `scss`, `sass` and
+`css` files for either `"sass"` or `"style"`.
+
+## Types
+
```

Comment:
````suggestion

```ts
import '../spec/js-api';
```

````

Asked ChatGPT about this:

> you can use an import assignment to make your ambient module declaration valid. This works because the import assignment tells TypeScript that the module "../spec/js-api" exists and has some type, and then the ambient module declaration augments that type with additional information.

So it looks like we need an import so TS knows where to find the module we will declare below?

Reply:
I wouldn't rely on anything ChatGPT tells you. You don't need anything more than the module declaration (see https://github.com/sass/sass/blob/main/accepted/prev-url.d.ts.md#types for example).

Reply:
@nex3 The module declaration for `prev-url` only compiles because of the import of `PromiseOr`. If you remove that import (and remove that as the return value) from `prev-url.d.ts` (or from the md version, and then tangle), you'll see the same issue. 

@jerivas's suggestion does silence the typescript compiler. I'm not sure if there would be side effects.

Reply:
Oh, interesting. In that case, having a bare import is probably fine.